pkg/machines/components/vm/vmExpandedContent.jsx

```diff
@@ -27,40 +27,65 @@ import VmNetworkTab from '../vmnetworktab.jsx';
 import Consoles from '../consoles.jsx';
 import VmOverviewTab from '../vmOverviewTabLibvirt.jsx';
 import VmUsageTab from './vmUsageTab.jsx';
+import VmSnapshotsTab from '../vmSnapshotsTab.jsx';
 import { ListingPanel } from 'cockpit-components-listing-panel.jsx';
 
 const _ = cockpit.gettext;
 
 /** One VM in the list (a row)
  */
-export const VmExpandedContent = ({
-    vm, vms, config, libvirtVersion, hostDevices, storagePools,
-    onUsageStartPolling, onUsageStopPolling, dispatch, networks, interfaces, nodeDevices, resourceHasError, onAddErrorNotification
-}) => {
-    const overviewTabName = (<a href="#" id={`${vmId(vm.name)}-overview`}>{_("Overview")}</a>);
-    const usageTabName = (<a href="#" id={`${vmId(vm.name)}-usage`}>{_("Usage")}</a>);
-    const disksTabName = (<a href="#" id={`${vmId(vm.name)}-disks`}>{_("Disks")}</a>);
-    const networkTabName = (<a href="#" id={`${vmId(vm.name)}-networks`}>{_("Network Interfaces")}</a>);
-    const consolesTabName = (<a href="#" id={`${vmId(vm.name)}-consoles`}>{_("Consoles")}</a>);
+export class VmExpandedContent extends React.Component {
+    constructor(props) {
+        super(props);
 
-    const tabRenderers = [
-        { name: overviewTabName, id: cockpit.format("$0-overview", vmId(vm.name)), renderer: VmOverviewTab, data: { vm, config, dispatch, nodeDevices, libvirtVersion } },
-        { name: usageTabName, id: cockpit.format("$0-usage", vmId(vm.name)), renderer: VmUsageTab, data: { vm, onUsageStartPolling, onUsageStopPolling }, presence: 'onlyActive' },
-        { name: disksTabName, id: cockpit.format("$0-disks", vmId(vm.name)), renderer: VmDisksTab, data: { vm, vms, config, storagePools, onUsageStartPolling, onUsageStopPolling, dispatch, onAddErrorNotification }, presence: 'onlyActive' },
-        { name: networkTabName, id: cockpit.format("$0-networks", vmId(vm.name)), renderer: VmNetworkTab, presence: 'onlyActive', data: { vm, dispatch, config, hostDevices, interfaces, networks, nodeDevices, onAddErrorNotification } },
-        { name: consolesTabName, id: cockpit.format("$0-consoles", vmId(vm.name)), renderer: Consoles, data: { vm, config, dispatch, onAddErrorNotification } },
-    ];
+        this.state = {
+            snapshotsAPIsAvailable: false,
```

Comment:
For a single bit of state, using a [React hook](https://reactjs.org/docs/hooks-overview.html) is nicer than rewriting the entire component as a class. Then you also don't need to do the `self.props.blah` addressing and the `{ long, list } = self.props` decomposition.

Your choice, though, I just want to point it out.

Reply:
Don't you also need stateful component in order to use componentDidMount ? Altough I could use useEffect.
I don't really have stance on this tho.

Reply:
Right, that'd be an Effect. But as I said, this is not a "change this!" comment, just a FYI.